### ISSUE TITLE
The trial won't run under Ubuntu18.04 when gpuNum is set to 1.

### ISSUE THREAD

**Issue by lizhaodong** [microsoft/nni#1600]
The trial won't run under Ubuntu18.04 when gpuNum is set to 1. The trail keeps waiting. When I set gpuNum to 0, it works normally.

**Short summary about the issue/question**:

**Brief what process you are following**: 

**How to reproduce it**: 
My own code didn't work. I tried on the mnist-keras example and it didn't work if gpuNum = 1.

**nni Environment**:
- nni version: v1.0
- nni mode(local|pai|remote):local 
- OS:ubuntu 18.04
- python version: 3.6
- is conda or virtualenv used?: no
- is running in docker?: no

**need to update document(yes/no)**:

**Anything else we need to know**:
see bug #1037，they are similar.

**Comment by lizhaodong** [comment]
~/nni/experiments/xmNdwRon: only exist checkpoint  db  log folders, not exist trails .
vim nnimanager.log: 
  1 [2019-10-10 18:07:43] INFO [ 'Datastore initialization done' ]
  2 [2019-10-10 18:07:43] INFO [ 'Rest server listening on: http://0.0.0.0:8080' ]
  3 [2019-10-10 18:07:43] INFO [ 'RestServer start' ]
  4 [2019-10-10 18:07:43] INFO [ 'Construct local machine training service.' ]
  5 [2019-10-10 18:07:43] INFO [ 'RestServer base port is 8080' ]
  6 [2019-10-10 18:07:45] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"command":"python3 mnist-keras.py","codeDir":"/home/lzd/nni/examples/trials    /mnist-keras","gpuNum":1}' ]
  7 [2019-10-10 18:07:45] INFO [ 'required GPU number is 1' ]
  8 [2019-10-10 18:07:45] INFO [ 'Starting experiment: xmNdwRon' ]
  9 [2019-10-10 18:07:45] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
 10 [2019-10-10 18:07:45] INFO [ 'Add event listeners' ]
 11 [2019-10-10 18:07:45] INFO [ 'Run local machine training service.' ]
 12 [2019-10-10 18:07:46] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
 13 [2019-10-10 18:07:46] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"optimizer": "Ad    am", "learning_rate": 0.0001}, "parameter_index": 0}' ]
 14 [2019-10-10 18:07:50] INFO [ 'submitTrialJob: form: {"jobType":"TRIAL","hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\",     \\"parameters\\": {\\"optimizer\\": \\"Adam\\", \\"learning_rate\\": 0.0001}, \\"parameter_index\\": 0}","index":0}}' ]

vim dispatcher.log:
  1 [10/10/2019, 06:07:46 PM] INFO (root/MainThread) Generating grammar tables from /usr/lib/python3.6/lib2to3/Grammar.txt
  2 [10/10/2019, 06:07:46 PM] INFO (root/MainThread) Generating grammar tables from /usr/lib/python3.6/lib2to3/PatternGrammar.txt
  3 [10/10/2019, 06:07:46 PM] INFO (nni.msg_dispatcher_base/MainThread) Start dispatcher
  4 [10/10/2019, 06:07:46 PM] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002915 seconds
  5 [10/10/2019, 06:07:46 PM] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials

**Comment by SparkSnail** [comment]
Hi, this problem might caused by that nni does not collect gpu_mterics file correctly. Could you please check if there is a nni_gpu_collector process when you start an experiment? You could use `ps -ef | grep nni` to check processes. Could you please check if there are some gpu metric files in /tmp/nni/script?

**Comment by lizhaodong** [comment]
Thank you!
exec "ps -ef | grep nni" result:
xx      25760 16423  0 10月09 pts/0  00:00:00 bash /tmp/nni/script/gpu_metrics_collector.sh
xx      25761 25760  0 10月09 pts/0  00:18:45 python3 -m nni_gpu_tool.gpu_metrics_collector
xx      30309 16423  1 09:53 pts/0    00:00:01 node /home/lzd/.local/nni/main.js --port 8080 --mode local --start_mode new
xx      30323 30309  0 09:53 pts/0    00:00:00 /bin/sh -c python3 -m nni --tuner_class_name TPE --tuner_args "{\"optimize_mode\":\"maximize\"}"
xx      30324 30323  3 09:53 pts/0    00:00:03 python3 -m nni --tuner_class_name TPE --tuner_args {"optimize_mode":"maximize"}
We can see  "nni_gpu_collector process" when I start an experiment. At same time,  there are some gpu metric files----"/tmp/nni/script/gpu_metrics_collector.sh " in /tmp/nni/script.

**Comment by lizhaodong** [comment]
This function maybe have a problem!
    public getAvailableGPUIndices(useActiveGpu: boolean, occupiedGpuIndexNumMap: Map<number, number>): number[] {
        if (this.gpuSummary !== undefined) {
            if (process.platform === 'win32' || useActiveGpu) {
                return this.gpuSummary.gpuInfos.map((info: GPUInfo) => info.index);
            } else {
                return this.gpuSummary.gpuInfos.filter((info: GPUInfo) =>
                         occupiedGpuIndexNumMap.get(info.index) === undefined && info.activeProcessNum === 0 ||
                         occupiedGpuIndexNumMap.get(info.index) !== undefined)
                       .map((info: GPUInfo) => info.index);
            }
        }

        return [];
    }

**Comment by lizhaodong** [comment]
look here:   info.activeProcessNum === 0,  but my gpu always contains system processes.

| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      5134      G   /usr/lib/xorg/Xorg                            18MiB |
|    0      9188      G   /usr/bin/gnome-shell                          58MiB |
|    0     10230      G   /opt/teamviewer/tv_bin/TeamViewer             16MiB |
|    0     16057      C   /usr/bin/python3                           10243MiB |
|    0     16443      G   /usr/lib/xorg/Xorg                           117MiB |
|    0     16577      G   /usr/bin/gnome-shell                         185MiB

**Comment by ultmaster** [comment]
Maybe you should set `useActiveGpu` to `true`. Refer to https://nni.readthedocs.io/en/latest/Tutorial/ExperimentConfig.html.

**Comment by NTU-P04922004** [comment]
@ultmaster 

Thanks for the update, now I can run NNI in full speed on a single GPU machine with the following configuration.

trial:
  ...
  gpuNum: 1
localConfig:
  ...
  useActiveGpu: true

**Comment by chicm-ms** [comment]
Fixed.

**Comment by wshzd** [comment]
> @ultmaster
> 
> Thanks for the update, now I can run NNI in full speed on a single GPU machine with the following configuration.
> 
> ```
> trial:
>   ...
>   gpuNum: 1
> localConfig:
>   ...
>   useActiveGpu: true
> ```
yes ,you are right
